### PR TITLE
SprintsController Tests

### PR DESCRIPTION
**Pull request by DavidKang**
We add a missing test for SprintController and also fix collision validation

### CODE REVIEW

File: app/controllers/sprints_controller.rb [hunk 1]
```diff
@@ -50,12 +50,12 @@ def destroy
 
   def start
     # Generate new sprint, paint burndown char and upload it to Trello
-    system 'trollolo burndown --new_sprint --plot-to-board --output=trollolo '\
+    Kernel.system 'trollolo burndown --new_sprint --plot-to-board --output=trollolo '\
```
Comment (Ana06) [inline app/controllers/sprints_controller.rb]: why is this needed and how is it related with writing a test?

File: app/controllers/sprints_controller.rb [hunk 2]
```diff
@@ -64,6 +64,7 @@ def start
       return
     end
 
+    flash[:success] = 'The new sprint was successfully generated'
```
Comment (Ana06) [inline app/controllers/sprints_controller.rb]: unrelated to the commit

File: spec/controllers/sprints_controller_spec.rb [hunk 3]
```diff
@@ -1,4 +1,89 @@
...
+    sign_in user
+  end
+
+  describe 'GET #index' do
+    before do
+      get :index
+    end
+
+    it 'returns http success' do
+      expect(response).to have_http_status(:success)
```
Comment (Ana06) [inline spec/controllers/sprints_controller_spec.rb]: this test is not doing anything :disappointed:

File: spec/controllers/sprints_controller_spec.rb [hunk 4]
```diff
@@ -1,4 +1,89 @@
...
+  end
+
+  describe 'POST #create' do
+    before do
+      post :create, params: { sprint: { number: 33, start_date: '2018-02-12',
+                                         end_date: '2018-02-23' } }
+    end
+
+    it { expect(response).to redirect_to(sprints_path) }
+    it { expect(Sprint.count).to eq(1) }
```
Comment (Ana06) [inline spec/controllers/sprints_controller_spec.rb]: I would check instead that the number of sprints was increased by 1 :thinking:

File: spec/controllers/sprints_controller_spec.rb [hunk 5]
```diff
@@ -1,4 +1,89 @@
...
+  end
+
+  describe 'GET #edit' do
+    before do
+      get :edit, params: { id: sprint.id }
```
Comment (Ana06) [inline spec/controllers/sprints_controller_spec.rb]: you are not editing anything!!! :no_mouth: instead of testing that it suceded, I would check that something was edited

File: spec/controllers/sprints_controller_spec.rb [hunk 6]
```diff
@@ -1,4 +1,89 @@
...
+
+    it { expect { update_action }.to change { sprint.reload.number }.from(32).to(34) }
+  end
+
+  describe 'DELETE #destroy' do
+    before do
+      delete :destroy, params: { id: sprint.id }
+    end
+
+    it { expect(Sprint.find_by(id: sprint.id)).to be_nil }
```
Comment (Ana06) [inline spec/controllers/sprints_controller_spec.rb]: I would use `to change(Sprint, :count).by(-1)` instead

File: spec/controllers/sprints_controller_spec.rb [hunk 7]
```diff
@@ -1,4 +1,89 @@
...
+    it { expect(response).to redirect_to(sprints_path) }
+  end
+
+  describe 'GET #destroy' do
```
Comment (Ana06) [inline spec/controllers/sprints_controller_spec.rb]: GET #destroy? :confused: what is that? :confused:

Comment (DavidKang) [reply spec/controllers/sprints_controller_spec.rb]: copy and paste :see_no_evil:

File: spec/controllers/sprints_controller_spec.rb (hunk 5 above)
Comment (DavidKang) [reply spec/controllers/sprints_controller_spec.rb]: `#edit` doesn't change anything, only render the edit view

**Comment by Ana06** [comment]
[P12: (3) Frontpage: Burndown Chart](https://trello.com/c/Gm0FGST1/567-p12-3-frontpage-burndown-chart)

File: app/controllers/sprints_controller.rb (hunk 1 above)
Comment (Ana06) [reply app/controllers/sprints_controller.rb]: `SprintController.expects(:system).with('trollolo burndown').returns('bla')`

or

`SprintController.any_instance.expects(:system).with('trollolo burndown').returns('bla')`

File: spec/controllers/sprints_controller_spec.rb (hunk 5 above)
Comment (Ana06) [reply spec/controllers/sprints_controller_spec.rb]: remove it :wink:
